### PR TITLE
Prevent multiple connects from resulting in conflicting calls to remote.push

### PR DESCRIPTION
**Pull request by redgeoff**
More details can be found at https://github.com/hoodiehq/hoodie.js/issues/283

### CODE REVIEW

File: src/lib/store/remote.js [hunk 1]
```diff
@@ -283,15 +284,24 @@ function hoodieRemoteStore(hoodie, options) {
...
     if (name) {
       remoteName = name;
     }
+
+    connectDefer = defer();
     remote.connected = true;
     remote.trigger('connect');
-    return remote.bootstrap().then(function() {
+    remote.bootstrap().then(function() {
       remote.push();
```
Comment (gr2m) [inline src/lib/store/remote.js]: that should be `return remote.push()`.

Can you add a test that checks that calling remote.connect twice doesn't bootstrap twice? I can help

**Comment by gr2m** [comment]
@redgeoff could you add the `return` statement and then stash the commits? We'll start to follow [angulars commit message styleguide](https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md) soon, your commit message would than be `fix(remote): prevent multiple calls on connect` if I got it right :)

**Comment by redgeoff** [comment]
Will do. I'll be in touch.

File: src/lib/store/remote.js (hunk 1 above)
Comment (redgeoff) [inline src/lib/store/remote.js]: I'll work on the test, but are you sure that the `return remote.push()` is needed? It wasn't there before.

Comment (gr2m) [inline src/lib/store/remote.js]: yeah, please add it. `remote.connect` should not resolve if remote.push fails.

**Comment by redgeoff** [comment]
I could use some help trying to write the test as I have no experience working with karma:
1. I'm guessing that I should add a testing statement to _specs/hoodie/remote.spec.js_ in the `describe('#connect')` block?
2. How would I then check for multiple calls to `remote.bootstrap`? It doesn't look like using the `isBootstrapping` flag will work and neither will monitoring the bootstrap:start event. Do I need to modify the `openConnectSpy`? If so, I could use some direction.
Thanks for the help!

**Comment by gr2m** [comment]
Sorry just checking in, I've a bit of a hard time to catch up, please give me a few more days, worst case early next week, to get back to you

**Comment by gr2m** [comment]
@redgeoff could you add me as contributor to your fork?

**Comment by redgeoff** [comment]
@gr2m sure, you've been added!

**Comment by gr2m** [comment]
This is good to be merged for my side. Can you squash the commits and rebase on master? I can also help if you're too busy

**Comment by redgeoff** [comment]
@gr2m sorry, I'm not sure what you want. Did you want me to create a new branch from master and commit all these changes in a single commit?

**Comment by robinboehm** [comment]
@redgeoff get some readings about Rebase and Interactive Rebase. :)

should be something like:
1. git pull master --rebase ( rebase on master ) 
   -->  get current master commits and apply all your commits on top of there it --> clear history without merge commits
2. git rebase -i HEAD~4
   --> Edit your commit history with keys **e** and **s**
   Simple Example: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html
3. check the result
4. git push -f (or add the correct remote e.g. redgeoff of you has added both)
   You have to overwrite your remote history

Something like this should work.

**Comment by gr2m** [comment]
@redgeoff happy to pair on it, I'm available for most of the next 6 hours.

**Comment by redgeoff** [comment]
@robinboehm thanks for the help! @gr2m did I do something wrong or is the problem with Travis CI?

**Comment by boennemann** [comment]
What is the status on this?

**Comment by redgeoff** [comment]
@boennemann and @gr2m I believe I rebased the commits and pushed them to the singleconnect branch properly, but the Travis CI build isn't working.

**Comment by gr2m** [comment]
@redgeoff that is our fault, not yours. We are currently refactoring hoodie.js that will fix the problem with tests & travis, it will take 2 more weeks though. But as soon as it's done, I'll happily merge your changes. Sorry to let you wait, and thanks for the support!

Until then, you can use your branch with the fix in your apps. Let me know if you need any help with that.

**Comment by gr2m** [comment]
fyi I'm working on incorporating this into current master, should get this done this week.

**Pull request closed without merging**